test/client/lease_executor_test.cpp

```diff
@@ -23,54 +23,155 @@
+        curve::mds::FileInfo* fileInfo = new curve::mds::FileInfo();
+        fileInfo->set_filestatus(curve::mds::FileStatus::kFileCreated);
+        response_.set_allocated_fileinfo(fileInfo);
+
+        EXPECT_CALL(curveFsService_, RefreshSession(_, _, _, _))
+            .WillRepeatedly(DoAll(SetArgPointee<2>(response_),
+                                  Invoke(MockRefreshSession)));
+    }
+
+ protected:
+    const char* kSvrAddr = "127.0.0.1:21001";
+
+    brpc::Server server_;
+    curve::mds::MockNameService curveFsService_;
+    curve::mds::ReFreshSessionResponse response_;
+
+    MDSClient mdsClient_;
+    IOManager4File io4File_;
+    UserInfo userInfo_;
+    FInfo fi_;
+    LeaseOption leaseOpt_;
+    LeaseSession lease_;
+};
+
+TEST_F(LeaseExecutorTest, TestStartFailed) {
+    fi_.fullPathName = "/TestStartFailed";
+
     {
-        LeaseOption leaseOpt;
-        leaseOpt.mdsRefreshTimesPerLease = 0;
+        leaseOpt_.mdsRefreshTimesPerLease = 100;
+        lease_.leaseTime = 0;
 
-        LeaseExecutor leaseExecutor(leaseOpt, userInfo, &mdsClient, &io4File);
+        LeaseExecutor exec(leaseOpt_, userInfo_, &mdsClient_, &io4File_);
+        ASSERT_FALSE(exec.Start(fi_, lease_));
+    }
 
-        LeaseSession lease;
-        lease.leaseTime = 1000;
-        ASSERT_FALSE(leaseExecutor.Start(fi, lease));
+    {
+        leaseOpt_.mdsRefreshTimesPerLease = 0;
+        lease_.leaseTime = 1000;
+
+        LeaseExecutor exec(leaseOpt_, userInfo_, &mdsClient_, &io4File_);
+        ASSERT_FALSE(exec.Start(fi_, lease_));
     }
 }
 
+TEST_F(LeaseExecutorTest, TestStartStop) {
+    PrepareMockService();
+
+    fi_.fullPathName = "/TestStartStop";
+    fi_.filestatus = FileStatus::Created;
+
+    leaseOpt_.mdsRefreshTimesPerLease = 1;
+    lease_.leaseTime = 5000000;
+
+    LeaseExecutor exec(leaseOpt_, userInfo_, &mdsClient_, &io4File_);
+    ASSERT_TRUE(exec.Start(fi_, lease_));
+
+    std::this_thread::sleep_for(std::chrono::seconds(20));
+
+    ASSERT_NO_FATAL_FAILURE(exec.Stop());
+}
+
+TEST_F(LeaseExecutorTest, TestMultiStop) {
+    PrepareMockService();
+
+    fi_.fullPathName = "/TestMultiStop";
+    fi_.filestatus = FileStatus::Created;
+
+    leaseOpt_.mdsRefreshTimesPerLease = 1;
+    lease_.leaseTime = 5000000;
+
+    LeaseExecutor exec(leaseOpt_, userInfo_, &mdsClient_, &io4File_);
+    ASSERT_TRUE(exec.Start(fi_, lease_));
+
+    std::this_thread::sleep_for(std::chrono::seconds(10));
+
+    ASSERT_NO_FATAL_FAILURE(exec.Stop());
+    ASSERT_NO_FATAL_FAILURE(exec.Stop());
+}
+
+TEST_F(LeaseExecutorTest, TestNoStop) {
+    PrepareMockService();
+
+    fi_.fullPathName = "/TestNoStop";
+    fi_.filestatus = FileStatus::Created;
+
+    leaseOpt_.mdsRefreshTimesPerLease = 1;
+    lease_.leaseTime = 5000000;
+
+    LeaseExecutor exec(leaseOpt_, userInfo_, &mdsClient_, &io4File_);
+    ASSERT_TRUE(exec.Start(fi_, lease_));
+
+    std::this_thread::sleep_for(std::chrono::seconds(10));
+
+    // does not explicit call exec.Stop()
```

Comment:
delete

Reply:
annotations here are used for  explanation